components/collator/src/comparison.rs

```diff
@@ -62,6 +71,118 @@ impl AnyQuaternaryAccumulator {
     }
 }
 
+#[inline(always)]
+fn in_inclusive_range16(i: u16, start: u16, end: u16) -> bool {
+    i.wrapping_sub(start) <= (end - start)
+}
+
+fn split_prefix_u16<'a, 'b>(
+    left: &'a [u16],
+    right: &'b [u16],
+) -> (&'a [u16], &'a [u16], &'b [u16]) {
+    let mut i = left
+        .iter()
+        .zip(right.iter())
+        .take_while(|(l, r)| l == r)
```

Comment:
nit: this is better written as `.find()` I think.

Reply:
I'll try that as a follow-up, but not trying non-mandatory changes right now to get this landed in time.

Reply:
Yeah, I looked at `find` a bit more now, and I don't see how it would improve this.

Reply:
Hmm. `.take_while(predicate).count()` is a more verbose way of writing `.find(!predicate)`, with the difference being that an unsuccessful find returns None instead of the length.


It is being used as an index though, so having it be the length in the None case is a useful property. personally I'd write it as `.find(..).unwrap_or(right.len())` but that then assumes left and right have the same size.

So maybe the code shouldn't change, but I think I'd like to see the variable get a better name, and perhaps a comment. The purpose of the take_while is not immediately obvious without a closer look.